### PR TITLE
Initial commit to add base level slave configuration

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
There is a lot here initially as I wanted to make sure this was a full fledged puppet module with tests included. Due to the nature of using the foreman-infra puppet modules (specifically the slave module) somethings required some customization. The biggest two being a script to encapsulate and simplify applying the puppet module (and ensuring all dependencies are properly configured for the apply to work) as well as how the fixtures are generated when testing as this is different than the normal fixture setup for puppet module testing.

**Comment by ehelms** [comment]
Could someone with write access to this repository please turn on travis-ci for it?

**Comment by ehelms** [comment]
Closed / reopened to kick off Travis

**Comment by ehelms** [comment]
Tests are passing and this is ready for review. Things to check:

1) The parts that are applicable to properly configuring a slave for the required QE environment
2) The overall puppet module structure (@stbenjam would you be willing to assist?)

File: manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class robottelo_slave () {
```
Comment (stbenjam) [inline manifests/init.pp]: I realize none of the classes are very complicated but I do like class headers (I mean these: https://github.com/Katello/puppet-qpid/blob/master/manifests/init.pp#L1-L48)  regardless, it reminds future authors to update them especially if things are parameterized later

File: manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+  include ::slave
+  include ::libvirt
```
Comment (stbenjam) [inline manifests/init.pp]: Should these two classes be requirements in the metadata.json?

Comment (ehelms) [inline manifests/init.pp]: I wasn't sure what to do about that given they live in foreman-inra/puppet/modules and don't have associated git repositories for them individually or their own metadata.

Comment (stbenjam) [inline manifests/init.pp]: Oh I see, no big deal then

**Comment by stbenjam** [comment]
Given the context the puppet looks good so :smile_cat: and/or ACK from me

**Comment by elyezer** [comment]
ACK

**Pull request merged**